modules/auction-manager/src/lib.rs

```diff
@@ -565,69 +571,91 @@ impl<T: Config> Pallet<T> {
-				// can be fixed by treasury council.
-				if let Some(bidder) = maybe_bidder.clone() {
-					let res = T::CDPTreasury::issue_debit(&bidder, bid_price, false);
-					if let Err(e) = res {
-						log::warn!(
-							target: "auction-manager",
-							"issue_debit: failed to issue stable {:?} to {:?}: {:?}. \
-							This is unexpected but should be safe",
-							bid_price, bidder, e
-						);
-						debug_assert!(false);
-					}
-				}
-
-				if collateral_auction.in_reverse_stage(stable_amount) {
-					// refund extra stable currency to recipient
-					let refund_amount = stable_amount
-						.checked_sub(collateral_auction.target)
-						.expect("ensured stable_amount > target; qed");
-					// it shouldn't fail and affect the process.
-					// but even it failed, just the winner did not get the refund amount. it can be
-					// fixed by treasury council.
-					let res = T::CDPTreasury::issue_debit(&collateral_auction.refund_recipient, refund_amount, false);
-					if let Err(e) = res {
-						log::warn!(
-							target: "auction-manager",
-							"issue_debit: failed to issue stable {:?} to {:?}: {:?}. \
-							This is unexpected but should be safe",
-							refund_amount, collateral_auction.refund_recipient, e
-						);
-						debug_assert!(false);
+		// if bid_price doesn't reach target, DEX will try trading with DEX to get better result.
+		if !collateral_auction.in_reverse_stage(bid_price) {
+			let default_swap_parital_path_list: Vec<Vec<CurrencyId>> = T::DefaultSwapParitalPathList::get();
+			let stable_currency_id = T::GetStableCurrencyId::get();
+
+			// iterator default_swap_parital_path_list to try swap until swap succeed.
+			for partial_path in default_swap_parital_path_list {
+				let partial_path_len = partial_path.len();
+
+				// check collateral currency_id and partial_path can form a valid swap path.
+				if partial_path_len > 0
+					&& collateral_auction.currency_id != partial_path[0]
+					&& partial_path[partial_path_len - 1] == stable_currency_id
+				{
+					let mut swap_path = vec![collateral_auction.currency_id];
+					swap_path.extend(partial_path);
+
+					// DEX must take a higher bid.
+					if bid_price
+						< T::DEX::get_swap_target_amount(&swap_path, collateral_auction.amount).unwrap_or_default()
+					{
+						// try swap collateral in auction with DEX to get stable.
+						if let Ok(stable_amount) = T::CDPTreasury::swap_exact_collateral_to_stable(
+							collateral_auction.currency_id,
+							collateral_auction.amount,
+							Zero::zero(),
+							&swap_path,
+							true,
+						) {
+							// swap successfully, will not deal.
+							should_deal = false;
+
+							// refund stable currency to the last bidder, it shouldn't fail and affect the
+							// process. but even it failed, just the winner did not get the bid price. it
+							// can be fixed by treasury council.
+							if let Some(bidder) = maybe_bidder.clone() {
```

Comment:
Looks like `maybe_bidder` is not used anymore so clone is needed? Otherwise can also use `.as_ref`